### PR TITLE
[TASK] Add gateway filter

### PR DESCRIPTION
**Pull request by xf-**
## Description

Add gateway filter like suggested by @plumpudding 

## Checklist:

- [x] My code follows the code style of this project. (CI will test it anyway and also needs approval)

### CODE REVIEW

File: lib/proportions.js [hunk 1]
```diff
@@ -107,16 +109,34 @@ define(['d3-interpolate', 'virtual-dom', 'filters/genericnode', 'helper'],
           return _.t('node.deactivated');
         });
 
+        nodes.forEach(function (node) {
+          if (node.flags.gateway) {
+            var mesh = node.nodeinfo.network.mesh;
+            mesh[Object.keys(mesh)[0]].interfaces.tunnel.forEach(function (mac) {
```
Comment (genofire) [inline lib/proportions.js]: i do not like this kind of mapping in the object - but it will work

**Review by genofire — APPROVED** [review]

File: lib/proportions.js (hunk 1 above)
Comment (xf-) [reply lib/proportions.js]: static config vs this solution. some communities use mzbat or bat0 or... Only way to select the first object.

Comment (xf-) [reply lib/proportions.js]: don't merge  - if we add gw in detail - we need the same function

Comment (genofire) [reply lib/proportions.js]: I agree a static one is also bad.
Maybe there should a solution if there are more the one mesh interface.
I am not sure if it is better then they are three loops nested.

Possible this is already the best way.

Comment (xf-) [reply lib/proportions.js]: i tried multiple communities and always one device, but different names.

Comment (genofire) [reply lib/proportions.js]: okay then you got my okay to merge it

Comment (xf-) [reply lib/proportions.js]: #35 are similar information. maybe i will move the code to a helper function

Comment (genofire) [reply lib/proportions.js]: On a way, that it would not run it so often?

Comment (xf-) [reply lib/proportions.js]: "easiest" would be to rewrite the stats and also avoid some copy stuff like like all nodes to router and building a new index. (we have nodes on multiple places via setData doing similar stuff.) One data processor for all initial needed info (maybe move some tasks to yanic to reduce the work and heavy structure. e.g. in this example - i don't care for the name like mzbat or bat0)

**Pull request closed without merging**